api/src/main/java/org/openmrs/hl7/impl/HL7ServiceImpl.java

```diff
@@ -902,9 +902,10 @@ private void setFatalError(HL7InQueue hl7InQueue, String error, Throwable cause)
 			hl7InError.setErrorDetails("");
 		} else {
 			StringWriter sw = new StringWriter();
-			PrintWriter pw = new PrintWriter(sw, true);
-			cause.printStackTrace(pw);
-			pw.flush();
+			//			PrintWriter pw = new PrintWriter(sw, true);
+			//			cause.printStackTrace(pw);
+			log.error(cause);
+			//			pw.flush();
 			sw.flush();
 			hl7InError.setErrorDetails(OpenmrsUtil.shortenedStackTrace(sw.toString()));
```

Comment:
Please use hl7InError.setErrorDetails(ExceptionUtils.getStackTrace(case)); instead and get rid off StringWriter entirely. It's fine to keep log.error(cause).